src/services/axiosInstance.ts

```diff
@@ -0,0 +1,22 @@
+import axios, {AxiosResponse, InternalAxiosRequestConfig} from 'axios';
+import {baseUrl} from '@constants/api';
+
+const axiosInstance = axios.create({
+  baseURL: baseUrl,
+  timeout: 10000,
+  headers: {
+    'Content-Type': 'application/json'
+  }
+});
+
+axiosInstance.interceptors.response.use((response: AxiosResponse<any>) => response.data);
+
+axiosInstance.interceptors.request.use((config): InternalAxiosRequestConfig<any> => {
+  const accessToken = window.localStorage.getItem('accessToken');
+  if (accessToken) {
```

Comment:
What about the case if token is expired?

Reply:
Then I need refresh token. I have such api at the backend but it is not here yet. I was not sure if is related to this pull request, but I can do it here I think.

Reply:
By the way I also have a question where it is better to fetch get current user api. Because I don`t have an App.tsx where I do it normally as I use the router provider inside main.tsx... Maybe ask you tomorrow if I couldn`t find solution.

Reply:
I have a question related to App.tsx and where to call a get user api for the first time. What if I do it in a MainLayout component? I realise that layouts are not for api calls, but with my router structure MainLayout is close to App. Or don`t do that and as we mentioned just create a separate App.tsx? Thanks. 
```
export const router = createBrowserRouter([
  {
    element: <MainLayout />,
    children: [
      {
        path: `${routes.home}`,
        element: <HomePage />
      },
      {
        path: '*',
        element: <NotFoundPage />
      }
    ]
  }
]);
```